src/stats/inc/LikelihoodBase.h

```diff
@@ -77,10 +85,31 @@ class BaseGaussianLikelihood : public BaseScalarFunction<V, M> {
    * the observational covariance matrix.  In this case, the user need not
    * concern themselves with this parameter as it is handled not in the model
    * evaluation but by the likelihood evaluation.
+   *
+   * By default, to ensure backward compatibility, this method will call the
+   * deprecated
+   * evaluateModel( const V & domainVector, const V * domainDirection,
+   * V & modelOutput, V * gradVector, M * hessianMatrix, V * hessianEffect ).
    */
-  virtual void evaluateModel(const V & domainVector, const V * domainDirection,
-      V & modelOutput, V * gradVector, M * hessianMatrix,
-      V * hessianEffect) const = 0;
+  virtual void evaluateModel(const V & domainVector, V & modelOutput) const
+  { this->evaluateModel(domainVector,NULL,modelOutput,NULL,NULL,NULL); }
+
+  //! Logarithm of the value of the scalar function.
+  /*!
+   * This interface is deprecated. Subclasses should instead implement
+   * double lnValue(const V & domainVector)
+   */
+  virtual double lnValue(const V & domainVector, const V * domainDirection,
+      V * gradVector, M * hessianMatrix, V * hessianEffect) const;
+
+  //! Actual value of the scalar function.
+  virtual double actualValue(const V & domainVector, const V * domainDirection,
+                             V * gradVector, M * hessianMatrix, V * hessianEffect) const
+  { return std::exp(this->lnValue(domainVector, domainDirection, gradVector,
+                                  hessianMatrix, hessianEffect)); }
+
+  //! Logarithm of the value of the scalar function.
+  virtual double lnValue(const V & domainVector) const =0;
```

Comment:
1. Oops, copy/paste error, I'll fix it (remove the pure part). I still like to have the virtual keyword there to clue us in that this may be overriding a function. Thanks for catching this.

2. Because I was thinking about `evaluateModel` backward compatibility and mistakenly applied that logic to `lnValue` as well. Thank you for pointing this out.